### DISCUSSION TITLE
esp32 wlan sta connect with wrong password sta.status return 205

### DISCUSSION THREAD

**Discussion by puppet13th** [micropython/micropython#9572] [Hardware & Peripherals] [answered]
when connecting sta with wrong password ,sta.status() should return 202, but it return 205:
205 does not found in network module: 
```
  STAT_IDLE -- 1000
  STAT_CONNECTING -- 1001
  STAT_GOT_IP -- 1010
  STAT_NO_AP_FOUND -- 201
  STAT_WRONG_PASSWORD -- 202
  STAT_BEACON_TIMEOUT -- 200
  STAT_ASSOC_FAIL -- 203
  STAT_HANDSHAKE_TIMEOUT -- 204
```
firmware : MicroPython v1.19.1-528-gb8982ec5f on esp32-c3

**Comment by IhorNehrutsa** [comment, marked as answer]
```
 WIFI_REASON_CONNECTION_FAIL                    = 205,
```
see https://github.com/espressif/esp-idf/blob/c8243465e45489835d645bf217a6929fd0c01b7f/components/esp_wifi/include/esp_wifi_types.h#L131

**esp32/network_wlan: Fix network.WLAN.status().** #12932
In this PR, when an incorrect password causes WIFI_REASON_CONNECTION_FAIL(205), then sta.status() returns STAT_WRONG_PASSWORD.

**Reply by ctimmer** [reply]
Is this fixed on all ports? I had a similar problem on the picow (v1.21.0) except it just looped on an invalid password.

**Reply by IhorNehrutsa** [reply]
esp32 only

**Reply by ctimmer** [reply]
That's disappointing. There must be a lot of common code between the different ports. Don't the developers share issues? Do common problems have to be reported on each port?

Don't get me wrong, I think you guys do a great job.

Curt
